ui/app/appointments/directives/datePicker.js

```diff
@@ -22,18 +23,24 @@ angular.module('bahmni.appointments')
                 $scope.viewDate = $scope.viewDate && dateUtil.addDays($scope.viewDate, 1);
             };
 
-            $scope.$watch("viewDate", $scope.onChange($scope.viewDate));
-
+            $scope.$watch("viewDate", function (viewDate) {
+                $scope.viewDate = viewDate;
+                if (!viewDate) {
```

Comment:
Hi @smartyrad ,
You can simplify the if check. Try to do `$scope.viewDate = viewDate || $scope.lastValidDate`

Reply:
@sumanmaity112 Good suggestion, I have implemented your changes and updated this PR